packages/commerce-sdk-react/src/hooks/ShopperProducts/query.ts

```diff
@@ -58,9 +106,27 @@ parameter. The server only returns online categories.
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperproducts.shopperproducts-1.html#getcategory} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useCategory = (
-    arg: Argument<Client['getCategory']>
-): QueryResponse<DataType<Client['getCategory']>> => {
+function useCategory(
+    parameters: NonNullable<Argument<Client['getCategory']>>['parameters'],
+    deps?: unknown[],
+    headers?: NonNullable<Argument<Client['getCategory']>>['headers'],
+    rawResponse?: false
+): QueryResponse<DataType<Client['getCategory']>>
```

Comment:
Minor: what if we refactor the repeating complex types like this? So the code is a bit more readable. And we can see easier which inputs and outputs differ in the overloading functions.

```suggestion
function useCategory(
    parameters: Parameters,
    deps?: unknown[],
    headers?: Headers,
    rawResponse?: false
): QueryResponse<DataType<Client['getCategory']>>
```